tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,196 @@
+            'extensions': '[]'
+            },
+        'ch_event': {
+            'abs_change': 'Not specified',
+            'extensions': '[]',
+            'rel_change': 'Not specified'
+            },
+        'per_event': {
+            'extensions': '[]',
+            'period': '1000'
+            }
+        },
+    'format': '%s',
+    'label': 'Status',
+    'max_alarm': 'Not specified',
+    'min_alarm': 'Not specified',
+    'model': 'monctl:10000/sys/database/2/Status',
+    'name': 'Status',
+    'polling': 0,
+    'quality': 'ATTR_VALID',
+    'standard_unit': 'No standard unit',
+    'string': 'Device is OK',
+    'time': 1521724934.799581,
+    'unit': '',
+    'value': 'Device is OK',
+    'writable': 'READ'
+    }
+
+
+class GenericSetup(unittest.TestCase):
+    """A class providing the setUp method definition for the other test classes.
+    """
+    longMessage = True
+
+    def setUp(self):
+        super(GenericSetup, self).setUp()
+        self.fandango_json_file = pkg_resources.resource_filename(
+            'tango_simlib.tests.config_files', 'database2.json')
+        self.fandango_parser = fandango_json_parser.FandangoExportDeviceParser()
+        self.fandango_parser.parse(self.fandango_json_file)
+
+
+class TestFandangoJsonParser(GenericSetup):
+    """A test class that tests that the fandango json parser works correctly.
+    """
+    def test_preprocess_command_types(self):
+        """Testing that command property mapping was done well and command type
+        of mapped properties is of tango type.
+        """
+
+        # testing that the command properties have been renamed to
+        # match values in command signature
+        for cmd_name in self.fandango_parser.get_device_command_metadata():
+            self.assertEquals(
+                EXPECTED_CMD_PROPERTY_PARAMETERS.sort(),
+                self.fandango_parser.get_device_command_metadata()
+                [cmd_name].keys().sort(),
+                "The mapping of the command properties to the command"
+                "signature wasn't successful for '%s' command" % (cmd_name)
+                )
+            # testing that data type of cmd property values is a tango type and
+            # there is no occurence of 'Const' in the value of the cmd property
+            for cmd_prop_value in [
+                    self.fandango_parser.get_device_command_metadata()
+                    [cmd_name][INTERESTED_CMD_PROP[0]],
+                    self.fandango_parser.get_device_command_metadata()
+                    [cmd_name][INTERESTED_CMD_PROP[1]]]:
+                self.assertEquals(
+                    str(cmd_prop_value).find('Const'), -1,
+                    "The value (%s) of the cmd property, '%s', for the command, '%s',"
+                    "has 'Const'" % (cmd_prop_value, INTERESTED_CMD_PROP[0], cmd_name)
+                    )
+                self.assertEquals(
+                    EXPECTED_TANGO_TYPES[0], type(cmd_prop_value),
```

Comment:
Similarly to the `INTERESTED_CMD_PROPS`, we aren't using the `EXPECTED_TANGO_TYPES` list generically (i.e. just iterating through it), so using `tango._tango.CmdArgType` here directly, would be clearer for me.

Reply:
noted